src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/IQueueItem.cs

```diff
@@ -20,19 +19,27 @@ internal interface IQueueItem<TRequestContext>
     /// <summary>
     /// Executes the work specified by this queue item.
     /// </summary>
-    /// <param name="requestContext">the context created by <see cref="CreateRequestContextAsync(IMethodHandler, CancellationToken)"/></param>
+    /// <param name="request">The request parameters.</param>
+    /// <param name="context">The context created by <see cref="CreateRequestContextAsync{TRequest}(TRequest, IMethodHandler, CancellationToken)"/>.</param>
     /// <param name="handler">The handler to use to execute the request.</param>
+    /// <param name="language">The language for the request.</param>
     /// <param name="cancellationToken" />
     /// <returns>A <see cref="Task "/> which completes when the request has finished.</returns>
-    Task StartRequestAsync(TRequestContext requestContext, IMethodHandler handler, CancellationToken cancellationToken);
+    Task StartRequestAsync<TRequest, TResponse>(TRequest request, TRequestContext? context, IMethodHandler handler, string language, CancellationToken cancellationToken);
 
     /// <summary>
     /// Creates the context that is sent to the handler for this queue item.
     /// Note - this method is always called serially inside the queue before
-    /// running the actual request in <see cref="StartRequestAsync(TRequestContext, IMethodHandler, CancellationToken)"/>
+    /// running the actual request in <see cref="StartRequestAsync{TRequest, TResponse}(TRequest, TRequestContext?, IMethodHandler, string, CancellationToken)"/>
     /// Throwing in this method will cause the server to shutdown.
     /// </summary>
-    Task<TRequestContext> CreateRequestContextAsync(IMethodHandler handler, CancellationToken cancellationToken);
+    Task<TRequestContext> CreateRequestContextAsync<TRequest>(TRequest deserializedRequest, IMethodHandler handler, CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Attempts to deserialize the request.  If the request cannot be deserialized, ensures that the exception is returned as the result.
```

Comment:
> ensures that the exception is returned as the result.

Can you clarify what that means?  Is the exception trhown?  If it's returned... how?  How does it get embedded into TRequest?

Reply:
We set the exception on the completion source, so the client can observe it.

Reply:
sorry.  can you clarify *in the comment*.